core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java

```diff
@@ -49,33 +51,54 @@ public class TabletFile implements Comparable<TabletFile> {
 
   private static final Logger log = LoggerFactory.getLogger(TabletFile.class);
 
+  private static final String HDFS_TABLES_DIR_NAME = HDFS_TABLES_DIR.substring(1);
+
   /**
    * Construct new tablet file using a Path. Used in the case where we had to use Path object to
    * qualify an absolute path or create a new file.
    */
   public TabletFile(Path metaPath) {
     this.metaPath = Objects.requireNonNull(metaPath);
-    String errorMsg = "Missing or invalid part of tablet file metadata entry: " + metaPath;
+    final String errorMsg = "Missing or invalid part of tablet file metadata entry: " + metaPath;
```

Comment:
Normally this error message will not be needed. Would be good to avoid computing it.  Could have a static method that computes it to avoid duplicate code.  Everywhere its needed call the static method and pass in metaPath.